# Bonus: Temperature Analysis I

In [58]:
import pandas as pd
from datetime import datetime as dt

In [59]:
# "tobs" is "temperature observations"
df = pd.read_csv("Resources/hawaii_measurements.csv")

df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [67]:
# Convert the date column format from string to datetime
df['Date format'] = pd.to_datetime(df['date'], format='%Y-%m-%d')
df.head()

station        date  prcp  tobs Date format
0  USC00519397  2010-01-01  0.08    65  2010-01-01
1  USC00519397  2010-01-02  0.00    63  2010-01-02
2  USC00519397  2010-01-03  0.00    74  2010-01-03
3  USC00519397  2010-01-04  0.00    76  2010-01-04
4  USC00519397  2010-01-06   NaN    73  2010-01-06

In [68]:
# Set the date column as the DataFrame index
df.set_index(df['Date format'])
#Sort the dataframe by date
df = df.sort_index()

In [69]:
# Drop the date column
del df['date']
df.head()

station  prcp  tobs Date format
0  USC00519397  0.08    65  2010-01-01
1  USC00519397  0.00    63  2010-01-02
2  USC00519397  0.00    74  2010-01-03
3  USC00519397  0.00    76  2010-01-04
4  USC00519397   NaN    73  2010-01-06

In [72]:
df1 = df.
df1

station  prcp  tobs Date format
0      USC00519397  0.08    65  2010-01-01
1      USC00519397  0.00    63  2010-01-02
2      USC00519397  0.00    74  2010-01-03
3      USC00519397  0.00    76  2010-01-04
5      USC00519397  0.06    70  2010-01-07
...            ...   ...   ...         ...
19543  USC00516128  0.13    72  2017-08-17
19545  USC00516128  0.09    71  2017-08-19
19547  USC00516128  0.56    76  2017-08-21
19548  USC00516128  0.50    76  2017-08-22
19549  USC00516128  0.45    76  2017-08-23

[18103 rows x 4 columns]

### Compare June and December data across all years 

In [31]:
from scipy import stats

#### Filter data for desired months

In [97]:
df12 = df[(df['Date format'].dt.month == 12)] 
dfDec = df12.dropna(how='any')
dfDec

station  prcp  tobs Date format
305    USC00519397  0.04    76  2010-12-01
306    USC00519397  0.00    74  2010-12-03
307    USC00519397  0.00    74  2010-12-04
308    USC00519397  0.00    64  2010-12-06
309    USC00519397  0.00    64  2010-12-07
...            ...   ...   ...         ...
19323  USC00516128  0.14    71  2016-12-27
19324  USC00516128  0.14    71  2016-12-28
19325  USC00516128  1.03    69  2016-12-29
19326  USC00516128  2.37    65  2016-12-30
19327  USC00516128  0.90    65  2016-12-31

[1405 rows x 4 columns]

In [98]:
df06 = df[(df['Date format'].dt.month == 0o6)] 
dfJun = df06.dropna(how='any')
dfJun

station  prcp  tobs Date format
133    USC00519397  0.00    78  2010-06-01
134    USC00519397  0.01    76  2010-06-02
135    USC00519397  0.00    78  2010-06-03
136    USC00519397  0.00    76  2010-06-04
137    USC00519397  0.00    77  2010-06-05
...            ...   ...   ...         ...
19492  USC00516128  0.02    79  2017-06-26
19493  USC00516128  0.10    74  2017-06-27
19494  USC00516128  0.02    74  2017-06-28
19495  USC00516128  0.04    76  2017-06-29
19496  USC00516128  0.20    75  2017-06-30

[1574 rows x 4 columns]

In [100]:
# Identify the average temperature for June
averageJun = dfJun.tobs.mean()
averageJun

74.88754764930114

In [101]:
# Identify the average temperature for December
averageDec = dfDec.tobs.mean()
averageDec

70.93024911032029

In [105]:
groupjun = dfJun.tobs
groupdec = dfDec.tobs
print(groupjun, groupdec)

133      78
134      76
135      78
136      76
137      77
         ..
19492    79
19493    74
19494    74
19495    76
19496    75
Name: tobs, Length: 1574, dtype: int64 305      76
306      74
307      74
308      64
309      64
         ..
19323    71
19324    71
19325    69
19326    65
19327    65
Name: tobs, Length: 1405, dtype: int64


In [106]:
# Create collections of temperature data
stats.ttest_ind(a=groupjun, b=groupdec, equal_var=True)
#(statistic=-0.6337, pvalue=0.53005)

Ttest_indResult(statistic=30.865349991562194, pvalue=9.8415346259008e-182)

In [ ]:
# Run paired t-test

### Analysis

The alpha level of a hypothesis test is the threshold we use to determine whether or not our p-value is significant enough to accept a null hypothesis. It is often set at 0.05 but it is sometimes set as low as 0.01 or as high as 0.10. Since this p-value 9.84 is higher than the alpha levels, we can accept the null hypothesis. A null hypothesis refers to a hypothesis that states that there is no relationship between two population parameters.

Thus, we don’t have sufficient evidence to say that the true mean temperature is not much different in June and December.